### PR TITLE
feature(mode): Add git awareness modes

### PR DESCRIPTION
**Pull request by pkyosx**
Why need this change ? / Root Cause :
--------------
We would like to add support on git awareness. 

Changed Made :
--------------
- An optional arguments --mode is introduced (default: all)
- --mode=all is the original behavior
- --mode=obey_gitignore will exclude files that is excluded by .gitignore
- --mode=staged_only will only check staged files

Test Scope / Change Impact :
--------------
- Add tests for three different modes

### CODE REVIEW

File: src/missing.py [hunk 1]
```diff
@@ -62,25 +69,102 @@ def _find_for_unittest(self, basedir: Optional[str] = None) -> bool:
...
+            return True
+        return path in self._tracked_files
+
+    def _find_tracked_files(self, mode):
+        if mode == 'obey_gitignore':
+            return self._list_files_obey_gitignore()
+        elif mode == 'staged_only':
+            return self._list_files_staged_only()
+        else:
+            return None
```
Comment (cmj0121) [inline src/missing.py]: nit: how about define the enum or const string for these mode string?

File: src/missing.py [hunk 2]
```diff
@@ -62,25 +69,102 @@ def _find_for_unittest(self, basedir: Optional[str] = None) -> bool:
...
+
+    def _list_files_staged_only(self):
+        # list staged files
+        tracked_files = (
+            subprocess.check_output(
+                ['git', '--no-pager', 'diff', '--name-only', '--cached', '-z'],
+                encoding='utf-8',
+            )
```
Comment (cmj0121) [inline src/missing.py:120]: nit: I prefer not execute with absolute path, not just execute with `git`. But other package[0] seems use `git` too

[0]: https://github.com/gitpython-developers/GitPython/blob/main/git/cmd.py#L236

**Review by cmj0121 — APPROVED** [review]
Perfect.

File: src/missing.py (hunk 1 above)
Comment (pkyosx) [reply src/missing.py]: Good idea. I will use Enum.

**Pull request merged**